### PR TITLE
Toggle Joystick Panel

### LINKED ISSUE #448: Toggle Joystick Panel
## Motivation
Our old rqt controls GUI had a toggle for enabling/disabling joystick control. We need to transition this plugin to our Foxglove GUI.

## Requirements
- [x] Create a enable/disable Joystick toggle
- [x] When the Joystick is enabled start sending the appropriate controls to `/controls/desired_power`
    - [x] Right joystick
      - [x] Forward: +x, Backwards: -x, Left: +y, Right: -y
      - [x] Twist CCW: +Yaw, Twist CW: -Yaw
    - [x] Throttle
      - [x] Forward: +z, Backwards: -z
    - [x] Thumb joystick
      - [x] Forwards: +Pitch, Backwards: -Pitch, Right: +Roll, Left: -Roll (use power 0.5)
    - [ ] 2 Torpedos
      - [ ] Hold down right red striped button and press button (5) or button (6) to launch each torpedo

Notes:
- Desired power ranges from [-1, 1] (except for Pitch/Roll)
- Messages should be published at 20 Hz (refer to `ThrusterSpeedsPanel.tsx` in branch `thruster-speeds-panel` to see how to continuously publish in an extension).
- See [this issue](https://github.com/DukeRobotics/robosub-ros/issues/456) for more information on how to interface with the joystick

### PR DESCRIPTION
**Pull request by maxwellmlin**
Closes #448.

Code to read buttons to launch/fire torpedos is included. The service call to actually fire will be included in a future PR once torpedoes are added to Oogway.

### CODE REVIEW

File: foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx
Comment (vedarshshah) [inline foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx:191]: Add a comment before this function in the same format as the one for `queryJoystick`.

File: foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx
Comment (vedarshshah) [inline foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx:1]: - [ ] Use a consistent name for the variable denoting the previous state when `setState` is called. Currently, both `oldState` and `previousState` are used. If other extensions use the same name, use that. Otherwise, pick a name to use for this extension and be consistent.
- [ ] Add a comment in the place where the service call to fire the torpedoes will be added. A good place for this might be line 254, just after the call to `publishPower`.

File: foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx
Comment (vedarshshah) [inline foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx:287]: Add a comment here explaining why this if statement is necessary.

File: foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx
Comment (vedarshshah) [inline foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx:33]: Add a blank line before the comment.

File: foxglove/README.md
Comment (vedarshshah) [inline foxglove/README.md:118]: Update all links in this section with the new Foxglove documentation.

**Review by vedarshshah — CHANGES_REQUESTED** [review]
- [ ] At the pool test, with a little twist of the joystick, the robot yawed too quickly. To mitigate this, scale the yaw power by `0.5`. In other words, the yaw power published should be half of what the joystick outputs.

File: foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx
Comment (maxwellmlin) [reply foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx:1]: By default, eslint errors warning comments (e.g., "TODO"). I added a new rule that changes this to only warn.

Comment (maxwellmlin) [reply foxglove/extensions/toggle-joystick-panel/src/ToggleJoystickPanel.tsx:1]: Going to use "prevState" for consistency with Sensors Status Panel. But should be noted that Thruster Speeds Panel uses "oldState".

**Comment by maxwellmlin** [comment]
@vedarshshah If time permits, I will test all changes on Sunday. If not, will need to wait until after break to merge.

**Review by vedarshshah — APPROVED** [review]
LGTM! Merge after testing the latest changes.

**Pull request merged**
